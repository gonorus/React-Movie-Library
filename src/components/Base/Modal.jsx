import React, { useEffect } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import './Modal.scss';

export const DisplayModal = (Component) => {
  const rooElement = document.getElementById('root');
  const modalContainer = document.createElement('div');
  modalContainer.className = 'modal-container';

  if (rooElement) {
    rooElement.appendChild(modalContainer);
    ReactDOM.render(Component, modalContainer);
  }
};

const Modal = ({ children }) => {
  useEffect(() => () => {
    const containerElement = document.getElementsByClassName('modal-container')[0];
    containerElement.remove();
  }, []);

  const OnCloseModal = () => {
    const containerElement = document.getElementsByClassName('modal-container')[0];
    if (containerElement) ReactDOM.unmountComponentAtNode(containerElement);
  };

  return (
    <div className='modal-dialog'>
      <button type='button' onClick={OnCloseModal}>Close</button>
      {children}
    </div>
  );
};
Modal.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};
export default Modal;
