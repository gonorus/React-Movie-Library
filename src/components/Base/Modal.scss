.modal-container {
  width: 100%;
  height: 100%;
  position: absolute;
  background-color: rgba(0,0,0,0.6);
  top: 0;
  left: 0;
  display: grid;
  place-content: center;

  .modal-dialog {
    background-color: white;
    padding: 8px;
    border-radius: .2em;
    display: grid;
    row-gap: 8px;

    button {
      align-self: center;
    }
  }
}