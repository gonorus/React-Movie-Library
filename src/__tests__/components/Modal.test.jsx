import React from 'react';
import renderer from 'react-test-renderer';
import Modal from '../../components/Base/Modal';

test('Modal Dialog displayed', () => {
  const component = renderer.create(
    <Modal>asd</Modal>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.unmount();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
