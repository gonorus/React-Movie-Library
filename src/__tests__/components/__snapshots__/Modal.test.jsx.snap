// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Dialog displayed 1`] = `
<div
  className="modal-dialog"
>
  asd
</div>
`;

exports[`Modal Dialog displayed 2`] = `null`;
